clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -792,12 +800,17 @@ private void timeoutCallsInFlight(TimeoutProcessor processor, Map<String, List<C
                 // only one we need to check the timeout for.
                 Call call = contexts.get(0);
                 if (processor.callHasExpired(call)) {
-                    log.debug("{}: Closing connection to {} to time out {}", clientId, nodeId, call);
-                    client.disconnect(nodeId);
-                    numTimedOut++;
-                    // We don't remove anything from the callsInFlight data structure.  Because the connection
-                    // has been closed, the calls should be returned by the next client#poll(),
-                    // and handled at that point.
+                    if (call.aborted) {
+                        log.warn("{}: aborted call {} is still in callsInFlight.", clientId, call);
```

Comment:
Hmm.  I'm having trouble finding a clear way to fail fast here.  The call has already been aborted-- aborting it again should not help.  We could try to initiate an adminclient close, but this is a very heavy hammer to use and might make the situation worse for the operator.